examples/allocation-endpoint/server/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Copyright 2022 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# syntax=docker/dockerfile:1
+
+FROM golang:1.16-alpine
+
+WORKDIR /app
+
+COPY go.mod ./
+COPY go.sum ./
+RUN go mod download
```

Comment:
I think you can copy just go.mod and do go mod tidy.

Reply:
Any reason to prefer go mod tidy?

Reply:
`go mod download` downloads the named modules into module cache whereas `go mod tidy` modifies the go.sum files based on source code and go.mod file as necessary. I'm not sure you can do `go mod download` without a go.sum file but you should be able to do `go mod tidy` without it.

Reply:
Thanks for the clarification. Yes, both works. So we can use either `go mod tidy` or `COPY go.sum ./`. I am still not sure if one is preferred over the other, but I am following the official [docker for golang example](https://docs.docker.com/language/golang/build-images/).